File: gome-android-client/src/main/java/com/guardanis/gome/commands/MouseMoveCommand.java
package com.guardanis.gome.commands;

import org.json.JSONException;
import org.json.JSONObject;

public class MouseMoveCommand implements Command {

    public enum MouseMode {
        MOVE("move"),
        SCROLL("scroll");

        private String type;
        MouseMode(String type){
            this.type = type;
        }
    }

    protected MouseMode mouseMode = MouseMode.MOVE;
    private int x;
    private int y;

    public MouseMoveCommand(float x, float y){
        this(MouseMode.MOVE, x, y);
    }

    public MouseMoveCommand(MouseMode mouseMode, float x, float y){
        this.mouseMode = mouseMode;
        this.x = (int) x;
        this.y = (int) y;
    }

    @Override
    public String getActionIdentifier() {
        return "mouse";
    }

    @Override
    public JSONObject toJson() throws JSONException {
        return new JSONObject()
                .put("type", mouseMode.type)
                .put("mouse_x", x)
                .put("mouse_y", y);
    }

}

File: gome-swift-client/gome/MouseMoveCommand.swift
import UIKit

class MouseMoveCommand: Command {
    
    private var type: String!
    private var dx: Int!
    private var dy: Int!
    
    init(_ type: String, dx: CGFloat, dy: CGFloat){
        self.type = type
        self.dx = Int(dx)
        self.dy = Int(dy)
    }
    
    func getActionId() -> String {
        return "mouse"
    }
    
    func getData() -> [String: Any] {
        var data = [String: Any]()
        data["type"] = type
        data["mouse_x"] = dx
        data["mouse_y"] = dy
        
        return data
    }
    
}
